dozer-ingestion/mysql/src/binlog.rs

```diff
@@ -210,15 +225,9 @@ impl BinlogIngestor<'_, '_, '_> {
 
                     let query = query_event.query_raw().trim_start();
 
-                    if query == b"BEGIN"
-                        && self
-                            .ingestor
-                            .handle_message(IngestionMessage::SnapshottingStarted)
-                            .await
-                            .is_err()
-                    {
-                        // If receiving side is closed, we can stop ingesting.
-                        return Ok(());
+                    if query == b"BEGIN" {
+                        transaction_position = prev_position;
```

Comment:
Is there a reason why there is a `transaction_position`?

Reply:
We need to start from transaction begin statement, because otherwise, we cannot get table map event. 
https://github.com/getdozer/dozer/blob/8dfcdffd9118e020232798da69092ce4f69d2c32/dozer-ingestion/mysql/src/binlog.rs#L528

Reply:
I see. Then it's more tied to a `TABLE_MAP_EVENT` than transaction begin?

Reply:
Yes. Of course, we can start from that event, but I think all ingestion messages from the same MySQL transaction need to have the same `position` in `state`.